### ISSUE TITLE
[Regression] IDE0031: False positive on .NET 11 Preview1

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn#82482]
**Version Used**: 
.NET 11 Preview 1 with VS Insiders 2026 [11513.90]

This is a regression. The problem does not occur using stable .NET 10 and stable VS 2026.

**Steps to Reproduce**:
A suggestion is offered to convert:
```c#
if (_session != null)
{
    _session.Disposed -= SessionOnDisposed;
    _session = null;
}
```
to:
```c#
_session?.Disposed -= SessionOnDisposed;
_session = null;
```

**Diagnostic Id**:
IDE0031: Null check can be simplified

**Expected Behavior**:
No suggestion to change the existing code.

**Actual Behavior**:
A suggestion is offered. We've increased the severity to Warning in Release builds. Combined with warnaserror, this breaks the build.

**Comment by CyrusNajmabadi** [comment]
I'm not understanding why this is being described as a false positive. This is the pattern that analyzer looks for.

**Comment by bkoelman** [comment]
Why is the changed code better? It simplifies the C# code, at the cost of lower runtime performance. It doesn't eliminate branching (in the context of [branch prediction in speculative execution](https://en.wikipedia.org/wiki/Speculative_execution)), while introducing an assignment that is no longer conditional.

**Comment by CyrusNajmabadi** [comment]
> Why is the changed code better? It simplifies the C# code,

That's exactly the purpose of it. 

If you don't like that, you can disable the rule. :-)

**Comment by bkoelman** [comment]
I like how the rule used to work, just not this new aspect. I wonder if suggestions should be offered that degrade runtime performance or introduce additional execution. I also find the resulting code less readable.

@stephentoub Would you consider my concern justified?

**Comment by CyrusNajmabadi** [comment]
>  I wonder if suggestions should be offered that degrade runtime performance 

Many rules do this. It's a non-goal for ide analyzers to have no runtime impact.  If that is problematic, the rule can be disabled or suppressed.  However, this is very by design.  

Note: the ide analyzers have not and likely never will be concerned with the cost of an assignment. 

You want far more strict analyzers for that.
